app/api/csv/csvExporter.ts

```diff
@@ -0,0 +1,217 @@
+import { Writable } from 'stream';
+import { EventEmitter } from 'events';
+import * as csv from '@fast-csv/format';
+import templates from 'api/templates';
+import { PropertySchema } from 'shared/types/commonTypes';
+import { TemplateSchema } from 'shared/types/templateType';
+import translate, { getLocaleTranslation, getContext } from 'shared/translate';
+import translations from 'api/i18n/translations';
+import formatters from './typeFormatters';
+
+export type SearchResults = {
+  rows: any[];
+  totalRows: number;
+  aggregations: {
+    all: {
+      _types: {
+        buckets: {
+          key: string;
+          // eslint-disable-next-line camelcase
+          doc_count: number;
+          // eslint-disable-next-line camelcase
+          filtered: { doc_count: number };
+        }[];
+      };
+    };
+  };
+};
+
+export type ExporterOptions = {
+  dateFormat: string;
+  language: string;
+};
+
+export type ExportHeader = {
+  label: string;
+  name: string;
+  common: boolean;
+};
+
+export type TemplatesCache = {
+  [id: string]: TemplateSchema;
+};
+
+export const getTypes = (searchResults: SearchResults, typesWhitelist: string[] = []) =>
+  typesWhitelist.length > 0
+    ? typesWhitelist
+    : searchResults.aggregations.all._types.buckets
+        .filter(bucket => bucket.filtered.doc_count > 0)
+        .map(bucket => bucket.key);
+
+const hasValue = (value: string) => value !== 'missing';
+
+export const getTemplatesModels = async (templateIds: string[]): Promise<TemplatesCache> =>
+  Promise.all(
+    templateIds.filter(hasValue).map(async (id: string) => templates.getById(id))
+  ).then(results =>
+    results.reduce<any>(
+      (memo, template) => (template ? { ...memo, [template._id]: template } : memo),
+      {}
+    )
+  );
+
+export const notDuplicated = (collection: any) => (item: any) =>
+  collection.findIndex((i: any) => Object.keys(i).every(key => i[key] === item[key])) < 0;
+
+export const excludedProperties = (property: PropertySchema) =>
+  !['geolocation', 'preview', 'markdown', 'nested'].includes(property.type);
```

Comment:
Does this mean that geolocation properties are also not included?  Or are they processed elsewhere?

Reply:
Answer: https://github.com/huridocs/uwazi/pull/2850#discussion_r412167988